### PR TITLE
[asyncio] bpo-30423: add regression test for orphan future causes "RuntimeError: Event loop stopped before Future completed."

### PR DESCRIPTION
**Pull request by jimmylai**
@1st1 
When comment out the changes (remove future.remove_done_callback(_run_until_complete_cb) out of finally block) in run_until_complete(), this test fails.

With the committed fix, this test can pass and it work as a regression test.

ERROR: test_run_until_complete_loop_orphan_future_close_loop (test.test_asyncio.test_base_events.BaseEventLoopTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jimmylai/workspace/cpython/Lib/test/test_asyncio/test_base_events.py", line 543, in test_run_until_complete_loop_orphan_future_close_loop
    loop.run_until_complete(foo(0.1))
  File "/Users/jimmylai/workspace/cpython/Lib/asyncio/base_events.py", line 463, in run_until_complete
    raise RuntimeError('Event loop stopped before Future completed.')
RuntimeError: Event loop stopped before Future completed.

https://bugs.python.org/issue30423

Ref: https://github.com/python/cpython/pull/1688

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

File: Lib/test/test_asyncio/test_base_events.py [hunk 1]
```diff
@@ -529,6 +529,20 @@ def test_run_until_complete_loop(self):
...
             other_loop.run_until_complete, task)
 
+    def test_run_until_complete_loop_orphan_future_close_loop(self):
+        async def foo(sec=0):
+            await asyncio.sleep(sec)
+        self.loop.close()
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        try:
+            with mock.patch('asyncio.base_events.BaseEventLoop.run_forever', side_effect=Exception):
```
Comment (1st1) [inline Lib/test/test_asyncio/test_base_events.py]: Make sure that all lines are < 80 chars per PEP 8

**Comment by 1st1** [comment]
LGTM but please fix code style

**Comment by asvetlov** [comment]
Codestyle is fixed

**Pull request merged**